### PR TITLE
Timestamp dev patch

### PR DESCRIPTION
**Pull request by ruck314**
### Description
- preformed `git merge origin/pre-release`
- Update LICENSE.txt for year 2020
- Add missing parameters to pyrogue.Root.__init__() #10
  - But this _Core.py is removed as part of timestamp-dev branch
- updating submodules locks
  - specifically updating lcls-timing-core lock to future v3.0.0 release
  - adding l2si-core lock to future v2.0.0 release
  - updating surf to future v2.0.4 release
- error checking for PGP_TYPE_G in firmware
- updating from axi.AxiStreamMonitoring to axi.AxiStreamMonAxiL in software
- changing arg from driverPath to dev in software

### CODE REVIEW

File: shared/ruckus.tcl [hunk 1]
```diff
@@ -9,9 +9,9 @@ if { [VersionCheck 2018.2 ] < 0 } {
 # Check for submodule tagging
 if { [info exists ::env(OVERRIDE_SUBMODULE_LOCKS)] != 1 || $::env(OVERRIDE_SUBMODULE_LOCKS) == 0 } {
    if { [SubmoduleCheck {axi-pcie-core}    {3.0.0}  ] < 0 } {exit -1}
-   if { [SubmoduleCheck {lcls-timing-core} {2.0.0}  ] < 0 } {exit -1}
-   if { [SubmoduleCheck {ruckus}           {2.0.1}  ] < 0 } {exit -1}
-   if { [SubmoduleCheck {surf}             {2.0.0}  ] < 0 } {exit -1}
+   if { [SubmoduleCheck {lcls-timing-core} {3.0.0}  ] < 0 } {exit -1}
+   if { [SubmoduleCheck {ruckus}           {2.0.4}  ] < 0 } {exit -1}
+   if { [SubmoduleCheck {surf}             {2.0.3}  ] < 0 } {exit -1}
```
Comment (bengineerd) [inline shared/ruckus.tcl]: We will need whichever surf release contains https://github.com/slaclab/surf/pull/567. I'm trying to verify it now and expect it to merge later today.

Comment (ruck314) [reply shared/ruckus.tcl]: Then that's going to be a future release of surf v2.0.4

Comment (ruck314) [reply shared/ruckus.tcl]: 044e522

**Comment by bengineerd** [comment]
What's the "Add missing parameters to pyrogue.Root.init() #10" about?

**Comment by ruck314** [comment]
```
            serverPort  = None,            # Zeromq server port.  Specify a port number to start a zeromq server.
            timeout     = 1.0,             # Communication timeout in seconds
```
Relating to propagating the `serverPort` & `timeout` arguments down the hierarchy

**Comment by bengineerd** [comment]
But I don't see anything related to that in the diff.

**Pull request merged**